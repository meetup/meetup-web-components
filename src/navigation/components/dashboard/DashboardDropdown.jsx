import React from 'react';
import PropTypes from 'prop-types';

import Flex from '../../../layout/Flex';
import FlexItem from '../../../layout/FlexItem';
import Modal from '../../../interactive/Modal';

export const PROFILE_CLASS = 'profileDropdown-content';
export const PROFILE_GROUP_LIST_ITEM_CLASS = 'profileDropdown-content-group';

/**
 * Creates dashboard dropdown used in navbar mobile implementation,
 * contains links to Pro Admin pages.
 * @param  {Object} labels and links to useful pages
 * @return {React.element} composed component
 */
export const DashboardDropdownComponent = ({ mobileTabs, dismissAction }) => {
	const {
		help,
		profile,
		logout,
		analytics,
		groups,
		events,
		templates,
		publicProfile,
		contact,
	} = mobileTabs;
	const { members } = mobileTabs || {};

	return (
		<Modal onDismiss={dismissAction}>
			<Flex justify="spaceBetween" className="align--left padding--all">
				<FlexItem growFactor={1} className="text--secondary margin--left">
					<ul className="list">
						<li className="list-item padding--bottom">
							<a href={analytics.link}>{analytics.label}</a>
						</li>
						{Boolean(members) && (
							<li className="list-item padding--bottom">
								<a href={members.link}>{members.label}</a>
							</li>
						)}
						<li className="list-item padding--bottom">
							<a href={groups.link}>{groups.label}</a>
						</li>
						<li className="list-item padding--bottom">
							<a href={events.link}>{events.label}</a>
						</li>
						<li className="list-item padding--bottom">
							<a href={templates.link}>{templates.label}</a>
						</li>
						<li className="list-item padding--bottom">
							<a href={publicProfile.link}>{publicProfile.label}</a>
						</li>
						<li className="list-item">
							<a href={profile.link}>{profile.label}</a>
						</li>
						<li className="padding--top">
							<a href={contact.link}>{contact.label}</a>
						</li>
						<li className="padding--top">
							<a href={help.link}>{help.label}</a>
						</li>
						<li className="padding--top">
							<a href={logout.link}>{logout.label}</a>
						</li>
					</ul>
				</FlexItem>
			</Flex>
		</Modal>
	);
};

DashboardDropdownComponent.propTypes = {
	mobileTabs: PropTypes.shape({
		analytics: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		members: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		groups: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		events: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		templates: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		publicProfile: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		profile: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		contact: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		help: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
		logout: PropTypes.shape({
			link: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
	}),
	dismissAction: PropTypes.func,
};

export default DashboardDropdownComponent;
