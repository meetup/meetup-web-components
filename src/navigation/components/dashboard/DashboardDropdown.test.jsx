import React from 'react';
import { shallow } from 'enzyme';

import DashboardDropdown from './DashboardDropdown';

const MOCK_PROPS = {
	mobileTabs: {
		analytics: {
			link: 'meetup.com',
			label: 'Analytics',
		},
		members: {
			link: 'meetup.com',
			label: 'Members',
		},
		groups: {
			link: 'meetup.com',
			label: 'Groups',
		},
		events: {
			link: 'meetup.com',
			label: 'Events',
		},
		templates: {
			link: 'meetup.com',
			label: 'Templates',
		},
		publicProfile: {
			link: 'meetup.com',
			label: 'Public Profile',
		},
		profile: {
			link: 'meetup.com',
			label: 'Your profile',
		},
		contact: {
			link: 'meetup.com',
			label: 'Contact',
		},
		help: {
			link: 'meetup.com',
			label: 'Help',
		},
		logout: {
			link: 'meetup.com',
			label: 'Logout',
		},
	},
	dismissAction: jest.fn(),
};

describe('Profile Dropdown', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<DashboardDropdown {...MOCK_PROPS} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should match snapshot without members tab', () => {
		const mobileTabs = { ...MOCK_PROPS.mobileTabs };
		delete mobileTabs.members;
		const updatedProps = { ...MOCK_PROPS, mobileTabs };
		const wrapper = shallow(<DashboardDropdown {...updatedProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
