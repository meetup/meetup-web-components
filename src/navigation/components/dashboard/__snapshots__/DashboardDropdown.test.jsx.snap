// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Dropdown should match snapshot 1`] = `
<Modal
  onDismiss={[MockFunction]}
>
  <Flex
    className="align--left padding--all"
    justify="spaceBetween"
  >
    <FlexItem
      className="text--secondary margin--left"
      growFactor={1}
    >
      <ul
        className="list"
      >
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Analytics
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Members
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Groups
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Events
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Templates
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Public Profile
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            href="meetup.com"
          >
            Your profile
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Contact
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Help
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Logout
          </a>
        </li>
      </ul>
    </FlexItem>
  </Flex>
</Modal>
`;

exports[`Profile Dropdown should match snapshot without members tab 1`] = `
<Modal
  onDismiss={[MockFunction]}
>
  <Flex
    className="align--left padding--all"
    justify="spaceBetween"
  >
    <FlexItem
      className="text--secondary margin--left"
      growFactor={1}
    >
      <ul
        className="list"
      >
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Analytics
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Groups
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Events
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Templates
          </a>
        </li>
        <li
          className="list-item padding--bottom"
        >
          <a
            href="meetup.com"
          >
            Public Profile
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            href="meetup.com"
          >
            Your profile
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Contact
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Help
          </a>
        </li>
        <li
          className="padding--top"
        >
          <a
            href="meetup.com"
          >
            Logout
          </a>
        </li>
      </ul>
    </FlexItem>
  </Flex>
</Modal>
`;
